lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -682,47 +738,79 @@ export default function getAPIImplementation(): APIInterface {
             // and if the user goes through the forgot password flow, it will create
             // an account there and it will work fine cause there the email is also verified.
 
-            let isSignUpAllowed = await AccountLinking.getInstance().isSignUpAllowed({
-                newUser: {
-                    recipeId: "emailpassword",
-                    email,
-                },
-                isVerified: false,
-                tenantId,
-                userContext,
+            const mfaInstance = MultiFactorAuthRecipe.getInstance();
+            let session = await Session.getSession(options.req, options.res, {
+                sessionRequired: false,
+                overrideGlobalClaimValidators: () => [],
             });
 
-            if (!isSignUpAllowed) {
-                const conflictingUsers = await AccountLinking.getInstance().recipeInterfaceImpl.listUsersByAccountInfo({
-                    tenantId,
-                    accountInfo: {
+            if (session === undefined || mfaInstance === undefined) {
```

Comment:
whist this is correct, I really prefer if we break down the flows as per the slack message into clear paths, explaining each path in the comment along with examples of how the user can get there.

Reply:
Factor flow is implemented